argon2-jvm-nolibs/src/main/java/de/mkammerer/argon2/BaseArgon2.java

```diff
@@ -226,13 +229,50 @@ public HashResult hashAdvanced(int iterations, int memory, int parallelism, byte
 
         int result = Argon2Library.INSTANCE.argon2_hash(
                 jnaIterations, jnaMemory, jnaParallelism, password, new Size_t(password.length), salt, new Size_t(salt.length),
-                hash, new Size_t(hash.length), encoded, new Size_t(encoded.length), getType().getJnaType(), new JnaUint32(version.getJnaVersion())
+                hash, new Size_t(hash.length), encoded, new Size_t(encoded.length), getType().getJnaType(), version.getJnaType()
         );
         checkResult(result);
 
         return new HashResult(hash, Native.toString(encoded, ASCII));
     }
 
+    @Override
+    public byte[] rawHashAdvanced(int iterations, int memory, int parallelism, char[] password, Charset charset, byte[] salt, byte[] secret, byte[] associatedData) {
+        byte[] pwd = toByteArray(password, charset);
+        return rawHashAdvanced(iterations, memory, parallelism, pwd, salt, secret, associatedData, defaultHashLength, Argon2Version.NUMBER);
+    }
+
+    @Override
+    public byte[] rawHashAdvanced(int iterations, int memory, int parallelism, byte[] password, byte[] salt, byte[] secret, byte[] associatedData, int hashLength, Argon2Version version) {
+        int length = hashLength > 0 ? hashLength : defaultHashLength;
+        Argon2_context.ByReference context = buildContextReference(iterations, memory, parallelism,
+                length, password, salt, secret, associatedData, version);
+
+        int result = callLibraryContext(context);
+        wipeMemory(context);
+        checkResult(result);
+
+        return context.out.getByteArray(0, length);
+    }
+
+    @Override
+    public boolean verifyAdvanced(int iterations, int memory, int parallelism, char[] password, Charset charset, byte[] salt, byte[] secret, byte[] associatedData, byte[] rawHash) {
+        byte[] pwd = toByteArray(password, charset);
+        return verifyAdvanced(iterations, memory, parallelism, pwd, salt, secret, associatedData, defaultHashLength, Argon2Version.NUMBER, rawHash);
+    }
+
+    @Override
+    public boolean verifyAdvanced(int iterations, int memory, int parallelism, byte[] password, byte[] salt, byte[] secret, byte[] associatedData, int hashLength, Argon2Version version, byte[] rawHash) {
+        int length = hashLength > 0 ? hashLength : defaultHashLength;
```

Comment:
Why this check? I would prefer to let argon2 raise an error instead of hiding it from the caller by using default hash lengths.